docs/topics/auth/passwords.txt

```diff
@@ -236,3 +236,173 @@ from the ``User`` model.
+    ``'username', 'first_name', 'last_name', 'email'``.
+    Attributes that don't exist are ignored.
+
+    The maximum similarity the password can have, before it is rejected, can
+    be set with the ``max_similarity`` parameter, on a scale of 0 to 1.
+    A setting of 0 will cause all passwords to be rejected, whereas a setting
+    of 1 will cause it to only rejects passwords that are identical to an
+    attribute's value.
+
+
+.. class:: CommonPasswordValidator()
+
+    Validates whether the password is not a common password, by checking
+    against a list of 1000 most common password created by
+    `Mark Burnett <https://xato.net/passwords/more-top-worst-passwords/>`_.
+
+Integrating validation
+-----------------------
+There are a few functions in ``django.contrib.auth.password_validation`` that
+you can call from your own forms or other code to integrate password
+validation. This can be useful if you use custom forms for password setting,
+or if you have API calls that allow passwords to be set, for example.
+
+.. function:: django.contrib.auth.password_validation.validate_password(password, user=None)
+
+    Validates a password. If all validators find the password valid, returns
+    ``None``. If one or more validators reject the password, raises a
+    :exc:`~django.core.exceptions.ValidationError` with all the error messages
+    from the validators.
+
+    The user object is optional: if it's not provided, some validators may not
+    be able to perform any validation and will accept any password.
+
+.. function:: django.contrib.auth.password_validation.password_changed(password, user=None)
+
+    Informs all validators that the password has been changed. This can be used
+    by some validators, e.g. a validator that prevents password reuse. This
+    should be called once the password has been successfully changed.
+
+.. function:: django.contrib.auth.password_validation.password_validators_help_texts()
+
+    Returns a list of the help texts of all validators. These explain the
+    password requirements to the user.
+
+.. function:: django.contrib.auth.password_validation.password_validators_help_text_html()
+
+    Returns an HTML string with all help texts in an ``<ul>``. This is
+    helpful when adding password validation to forms, as you can pass the
+    output directly to the ``help_text`` parameter of a form field.
+
+Note that where the password is passed to one of these functions, this should
+always be the clear text password - not a hashed password.
+
+
+Writing your own validator
+--------------------------
+If Django's built-in validators are not sufficient, you can write your own
+password validators. Validators are fairly simple classes. They must implement
+two methods:
+
+* ``validate(self, password, user=None)``: validate a password. Return
+  ``None`` if the password is valid, or a string with an error message if the
+  password is not valid. You must be able to deal with ``user`` being
```

Comment:
I'm worried about returning a string to announce a security-sensitive condition.

It would be easy for a developer to get confused with the general `validate_password` function and write code like:

    class MyValidator(SomeValidator):
        def validate(self, password, user=None):
            try:
                super().validate(password, user)
            except ValidationError:
                # do stuff...
                raise

Of couse that wouldn't pass tests, but I believe the principle of least astonishement is very important for security-sensitive APIs and that the discrepancy between `validate_password` and `validate` is sub-optimal in this regard.